### PR TITLE
CLN: 29547 replace old string formatting 4

### PR DESCRIPTION
**Pull request by 3vts**
I splitted PR #31844 in batches, this is the fourth
For this PR I ran the command `grep -l -R -e '%s' -e '%d' -e '\.format(' --include=*.{py,pyx} pandas/` and checked all the files that were returned for `.format(` and changed the old string format for the corresponding `fstrings` to attempt a full clean of, [#29547](https://github.com/pandas-dev/pandas/issues/29547). I may have missed something so is a good idea to double check just in case

- [ x  ] tests added / passed
- [ x ] passes `black pandas`
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/io/excel/test_readers.py [hunk 1]
```diff
@@ -596,7 +596,7 @@ def test_read_from_file_url(self, read_ext, datapath):
             # fails on some systems
             import platform
 
-            pytest.skip("failing on {}".format(" ".join(platform.uname()).strip()))
+            pytest.skip(f"failing on {' '.join(platform.uname()).strip()}")
```
Comment (simonjayhawkins) [inline pandas/tests/io/excel/test_readers.py]: I think the complexity of the substitution (measured by number of expressions) could warrant a separate assignment for clarity

**Review by simonjayhawkins — APPROVED** [review]
Thanks @3vts lgtm. one comment. not blocker.

File: pandas/tests/io/excel/test_readers.py (hunk 1 above)
Comment (3vts) [reply pandas/tests/io/excel/test_readers.py]: I separated the expression

**Comment by WillAyd** [comment]
Thanks @3vts

**Pull request merged**